test/Chainweb/Test/Pact/RemotePactTest.hs

```diff
@@ -227,6 +230,35 @@ localContTest iot nio = testCaseSteps "local continuation test" $ \step -> do
         $ mkCont
         $ mkContMsg pid 1
 
+pollingConfirmDepth :: IO (Time Micros) -> IO ChainwebNetwork -> TestTree
+pollingConfirmDepth iot nio = testCaseSteps "poll confirmation depth test" $ \step -> do
+    cenv <- _getServiceClientEnv <$> nio
+
+    step "/send a transaction"
+    cmd1 <- firstStep
+    rks <- sending cid' cenv (SubmitBatch $ pure cmd1)
+
+    step "/poll for the transaction until it appears"
+
+    beforePolling <- getCurrentBlockHeight v cenv cid'
+    PollResponses _ <- pollingWithDepth cid' cenv rks (Just $ ConfirmationDepth 10) ExpectPactResult
+    afterPolling <- getCurrentBlockHeight v cenv cid'
+
+    -- we are checking that we have waited at least 10 blocks using /poll for the transaction
+    assertBool "the difference between heights should be no less than the confirmation depth" $ (afterPolling - beforePolling) >= 10
+  where
+    cid' = unsafeChainId 0
+    tx = "42"
+    firstStep = do
+      t <- toTxCreationTime <$> iot
+      buildTextCmd
+        $ set cbSigners [mkSigner' sender00 []]
+        $ set cbCreationTime t
+        $ set cbNetworkId (Just v)
+        $ set cbGasLimit 70000
+        $ mkCmd "nonce-cont-1"
+        $ mkExec' tx
+
```

Comment:
We should add coverage more than a singleton batch here

Reply:
Could you please elaborate? What's missing?

Reply:
There are some cases when there are several items in the request keys.

Reply:
Added one more transaction to the request keys.